packages/graphql-server-hapi/src/hapiApollo.ts

```diff
@@ -71,23 +77,44 @@ graphqlHapi.attributes = {
   version: '0.0.1',
 };
 
-function verifyPayload(payload, reply) {
-  if (!payload) {
-    return reply(createErr(500, 'POST body missing.'));
-  }
+function assignBuffer(method, payload, query, reply) {
+    switch ( method ) {
+        case 'get':
+            if (!query) {
+                return reply(createErr(500, 'GET query missing.'));
+            }
+            return reply(query);
+        case 'post':
+            if (!payload) {
+                return reply(createErr(500, 'POST body missing.'));
+            }
+            return reply(payload);
+        default:
+            return reply(createErr(405, 'Apollo Server supports only GET/POST requests.'));
+    }
+}
 
+function assignIsBatch(method, buffer, reply) {
   // TODO: do something different here if the body is an array.
   // Throw an error if body isn't either array or object.
-  reply(payload && Array.isArray(payload));
+
+  switch ( method ) {
+      case 'get':
+          return reply(false);
```

Comment:
Ack, Where?
i just think that opening batching for GET is going to get monstrous URLs..